MapboxSearch/sdk-common/src/main/java/com/mapbox/search/common/OsmIdUtils.kt

```diff
@@ -0,0 +1,33 @@
+package com.mapbox.search.common
+
+import android.util.Base64
+
+/**
+ * Utility class.
+ */
+public object OsmIdUtils {
+    /**
+     * Mapping of supported OSM types -- node, way, and relation
+     */
+    private val osmTypes = mapOf(0 to "n", 1 to "w", 4 to "r")
+
+    /**
+     * Converts the supplied POI ID to a Mapbox ID
+     * @param poiId POI ID to convert
+     * @return [String] a Mapbox ID
+     * @throws IllegalArgumentException when the POI ID is not of type node, way, or relation
+     */
+    public fun fromPoiId(poiId: Long): String {
+        val lastDigit = poiId % 10
+        val osmType = osmTypes[lastDigit.toInt()]
+
+        if (osmType != null) {
+            val originalId = poiId / 10
+            val mbxPoiId = "urn:mbxpoi-osm:$osmType$originalId"
+            val bytes = mbxPoiId.toByteArray(Charsets.UTF_8)
+            return Base64.encodeToString(bytes, Base64.DEFAULT)
+        }
+
+        throw IllegalArgumentException("Invalid POI ID $poiId")
```

Comment:
I don't think it makes sense to throw an exception here; we can simply return null instead. Even from the user's perspective, this would be very inconvenient to use:

```
val osmId = try {
   fromPoiId(id)
} catch(e: IllegalArgumentException) {
    null
}
```

Reply:
Changed in latest commit to branch.